starting detecting using mediapipe

olaib opened the pull request:
starting detecting poses on images using mediapipe

olaib committed:
starting detecting poses on image using mediapipe

olaib committed:
start working on video stream

detecting pose on video instead of image

gavrielk commented on main.py:
```diff
@@ -0,0 +1,96 @@
+            landmark_list=detected_frame.pose_landmarks,
+            connections=mp.solutions.pose.POSE_CONNECTIONS,
+            landmark_drawing_spec=DRAWING.DrawingSpec(
+                color=(250, 250, 250),
```
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.

gavrielk commented on main.py:
```diff
@@ -0,0 +1,96 @@
+                circle_radius=4
+            ),
+            connection_drawing_spec=DRAWING.DrawingSpec(
+                color=(0, 0, 255),
```
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.

gavrielk commented on main.py:
```diff
@@ -0,0 +1,96 @@
+        result = detect_pose(frame,POSE_INIT)
+        draw_pose_on_frame(frame,result,DRAWING)
+        cv2.imshow('Video After detecting the pose', frame)
+        if cv2.waitKey(1) & 0xFF == ord('q'):
```
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.

gavrielk commented on main.py:
```diff
@@ -0,0 +1,96 @@
+    cap.release()
+    cv2.destroyAllWindows()
+
+def main(argv=None) -> None:
```
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be

gavrielk commented on main.py:
```diff
@@ -0,0 +1,96 @@
+    cap = cv2.VideoCapture(video_path)
+
+    while cap.isOpened():
+        ret,frame = cap.read()
```
How many frames per second do you manage to support?

olaib replied:
maybe 30, I tried to limit the fps to maximum 30 using cv2 see cap.set(cv2.CAP_PROP_FPS, FPS).
I will think about more efficient way to get exactly 30

excellenteam-scaleup replied:
Great, also measure it in the raspberryPi because the performance will be very different than what you get on your PC

olaib committed:
Starting with pose detector

some fixes and start working on pose detector

olaib committed:
calculating angles

starting calculating and displaying angles of detected body parts on image

olaib committed:
some fixes

fixing the angles

olaib committed:
some fixes and adding bar progress for correction precentage

olaib committed:
Some fixes

olaib committed:
some fixes

some fixes and fixing the exercise correctness (:
adding two threads for testing two videos one for correct and one for incorrect movement

todo: move the postDetector.py functions to Weightleft class
